x-pack/metricbeat/module/gcp/metrics/metricset.go

```diff
@@ -352,3 +360,36 @@ func addHostFields(groupedEvents []KeyValuePoint) mapstr.M {
 	}
 	return hostRootFields
 }
+
+func (m *MetricSet) setOrganization(ctx context.Context) error {
+
+	// Initialize the Cloud Resource Manager service
+	srv, err := cloudresourcemanager.NewService(ctx, m.config.opt...)
+	if err != nil {
+		return fmt.Errorf("failed to create cloudresourcemanager service: %w", err)
+	}
+
+	// Get the project ancestor details
+	ancestryResponse, err := srv.Projects.GetAncestry(m.config.ProjectID, &cloudresourcemanager.GetAncestryRequest{}).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("failed to get project ancestors: %w", err)
+	}
+	if len(ancestryResponse.Ancestor) == 0 {
+		return fmt.Errorf("no ancestors found for project '%s'", m.config.ProjectID)
+	}
+	ancestor := ancestryResponse.Ancestor[len(ancestryResponse.Ancestor)-1]
+
+	if ancestor.ResourceId.Type == "organization" {
+		m.config.organizationID = ancestor.ResourceId.Id
+		orgReq := srv.Organizations.Get(fmt.Sprintf("organizations/%s", m.config.organizationID))
+
+		orgDetails, err := orgReq.Context(ctx).Do()
+		if err != nil {
+			return fmt.Errorf("failed to get organization details: %w", err)
+		}
+
+		m.config.organizationName = orgDetails.DisplayName
+	}
```

Comment:
We are logging the error as a warning for now, as we still want the process to continue rather than stop metric collection if the top ancestor is not an organization. So should we return an error to the caller in this case?